test/e2e/framework/dataplane.go

```diff
@@ -241,3 +241,22 @@ func GetGlobalnetEgressParams(egressIP GlobalEgressIPType) GlobalnetTestParams {
 		GlobalEgressIP:   egressIP,
 	}
 }
+
+func IsSingleNodeDeployment(sourceNode, destNode framework.NetworkPodScheduling,
```

Comment:
The function name is kind of a misnomer - it's really checking if a connectivity test can be run involving a non-gateway node so how about:

```suggestion
func CanExecuteNonGatewayConnectivityTest(sourceNode, destNode framework.NetworkPodScheduling,
```